File: LegoFinal/ClientApp/src/app/Models/PartsResponse.ts
export class PartsResponse {
  count: number;
  next: string;
  previous: object;
  results: PartsResult[];

}

export class PartsResult {
  id: number;
  inv_part_id: number;
  part: Part;
  color: Color;
  set_num: string;
  quantity: number;
  is_spare: boolean;
  element_id: string;
  num_sets: number;
}

export class Part {
  part_num: string;
  name: string;
  part_cat_id: number;
  part_url: string;
  part_img_url: string;
  external_ids: External_Ids;
  print_of: object;
}

export class External_Ids {
  BrickOwl: string[];
  LDraw: string[];
  BrickLink: string[];
  Brickset: string[];
  LEGO: string[];
}

export class Color {
  id: number;
  name: string;
  rgb: string;
  is_trans: boolean;
  external_ids: External_Ids1;
}

export class External_Ids1 {
  BrickLink: Bricklink;
  BrickOwl: Brickowl;
  LEGO: LEGO;
  Peeron: Peeron;
  LDraw: Ldraw;
}

export class Bricklink {

  ext_ids: number[];
  ext_descrs: string[][];
}

export class Brickowl {

  ext_ids: number[];
  ext_descrs: string[][];
}

export class LEGO {

  ext_ids: number[];
  ext_descrs: string[][];
}

export class Peeron {

  ext_ids: number[];
  ext_descrs: string[][];
}

export class Ldraw {

  ext_ids: number[];
  ext_descrs: string[][];
}

File: LegoFinal/Models/PartsResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace LegoFinal.Models
{

    public class PartsResponse
    {
        public int count { get; set; }
        public object next { get; set; }
        public object previous { get; set; }
        public PartsResult[] results { get; set; }
    }

    public class PartsResult
    {
        public int id { get; set; }
        public int inv_part_id { get; set; }
        public Part part { get; set; }
        public Color color { get; set; }
        public string set_num { get; set; }
        public int quantity { get; set; }
        public bool is_spare { get; set; }
        public string element_id { get; set; }
        public int num_sets { get; set; }
    }

    public class Part
    {
        public string part_num { get; set; }
        public string name { get; set; }
        public int part_cat_id { get; set; }
        public string part_url { get; set; }
        public string part_img_url { get; set; }
        public External_Ids external_ids { get; set; }
        public object print_of { get; set; }
    }

    public class External_Ids
    {
        public string[] BrickOwl { get; set; }
        public string[] LDraw { get; set; }
        public string[] BrickLink { get; set; }
        public string[] Brickset { get; set; }
        public string[] LEGO { get; set; }
    }

    public class Color
    {
        public int id { get; set; }
        public string name { get; set; }
        public string rgb { get; set; }
        public bool is_trans { get; set; }
        public External_Ids1 external_ids { get; set; }
    }

    public class External_Ids1
    {
        public Bricklink BrickLink { get; set; }
        public Brickowl BrickOwl { get; set; }
        public LEGO LEGO { get; set; }
        public Peeron Peeron { get; set; }
        public Ldraw LDraw { get; set; }
    }

    public class Bricklink
    {
        public int[] ext_ids { get; set; }
        public string[][] ext_descrs { get; set; }
    }

    public class Brickowl
    {
        public int[] ext_ids { get; set; }
        public string[][] ext_descrs { get; set; }
    }

    public class LEGO
    {
        public int[] ext_ids { get; set; }
        public string[][] ext_descrs { get; set; }
    }

    public class Peeron
    {
        public object[] ext_ids { get; set; }
        public string[][] ext_descrs { get; set; }
    }

    public class Ldraw
    {
        public int[] ext_ids { get; set; }
        public string[][] ext_descrs { get; set; }
    }
}
